### ISSUE TITLE
Resource Status - Deployment status

### ISSUE THREAD

**Issue by bmesfin24** [PHACDataHub/sci-portal#38]
- Query the status from Config Sync
- The kpt repo has some design notes that'll help

**Comment by vedantthapa** [comment]
@seanpoulter-foci 
My proposal for this was to use the [kubernetes plugin](https://backstage.io/docs/features/kubernetes/) in backstage and then [surface kubernetes components as part of an entity](https://backstage.io/docs/features/kubernetes/configuration#surfacing-your-kubernetes-components-as-part-of-an-entity). Before I take it on, just wanted to confirm if you guys already explored this route?

**Comment by seanpoulter-foci** [comment]
It's an interesting idea. It supports [`customResources`](https://backstage.io/docs/features/kubernetes/configuration/#customresources-optional) but I haven't found where we'd be able to write a status query for the `Claim`s or other resources.

**Comment by vedantthapa** [comment]
@seanpoulter-foci

Yeah! This is how it'd look like in practice (ignore the permission error for pods) -
<img width="3008" alt="Screenshot 2024-06-05 at 11 07 44 AM" src="https://github.com/PHACDataHub/sci-portal/assets/43611693/d25eb55f-6388-4e83-9757-e5d424acc999">

Some details regarding the setup -
- It uses a k8s service account and some custom RBAC to authenticate and make requests.
- It prints out the entire `Status` field of the claim. If required, we can probably write [individual resource status to the claim's status](https://github.com/PHACDataHub/infra-core/blob/main/k8s/infra/internal/composites/xproject.yaml#L203-L214) so there's visibility into underlying resources.
- It is tagged to a `Resource` entity so only claims specific to that entity should show up on the page.
- https://github.com/PHACDataHub/sci-portal/pull/400 contains the changes for a claim and it's catalog-info.

Do you think this fits the use-case? / If we can build something custom on the frontend using this to surface the status?

This was really low-effort so if you think it doesn't fit the use-case here, I'm totally cool to discard it.

Big thanks @ssubedir for helping me set this up.
cc: @sleepycat, @tcaky
